src/dtos/http_response.go

```diff
@@ -0,0 +1,58 @@
+package dtos
+
+import (
+	"time"
+
+	"github.com/Real-Dev-Squad/wisee-backend/src/models"
+)
+
+type ErrorResponse struct {
+	Message string      `json:"message"`
+	Detail  interface{} `json:"detail"`
+}
+
+type ResponseDto struct {
+	Message string        `json:"message"`
+	Data    interface{}   `json:"data"`
+	Error   ErrorResponse `json:"error"`
+}
+
+type CreateUpdateGetFormResponseDto struct {
+	Id          int64              `json:"id"`
+	Content     models.FormContent `json:"content"`
+	OwnerId     int64              `json:"owner_id"`
+	CreatedById int64              `json:"created_by_id"`
+	Status      string             `json:"status"`
+	UpdatedById *int64             `json:"updated_by_id"`
+	CreatedAt   string             `json:"created_at"`
+	UpdatedAt   string             `json:"updated_at"`
+}
+
+type GetFormMetaDataResponseDto struct {
+	Id                               int64     `json:"id"`
+	FormId                           int64     `json:"form_id"`
+	IsDeleted                        bool      `json:"is_deleted"`
+	AccepctingResponses              bool      `json:"accepting_responses"`
+	AllowGuestResponses              bool      `json:"allow_guest_responses"`
+	AllowMultipleRepsonses           bool      `json:"allow_multiple_responses"`
+	SendConfirmationEmailToRespondee bool      `json:"send_confirmation_email_to_respondee"`
+	SendSubmissionEmailToOwner       bool      `json:"send_submission_email_to_owner"`
+	ValidTill                        time.Time `json:"valid_till"`
+	UpdatedById                      *int64    `json:"updated_by_id"`
+	// TODO invite code
+	UpdatedAt time.Time `json:"updated_at"`
+}
+
+type GetFormsResponseDto []CreateUpdateGetFormResponseDto
+
+type GetFormDetailResponseDto struct {
```

Comment:
- will `FormDetailsResponseDto` be a more suitable name here? as will allow us to use it in non`GET` routes as well?

Reply:
so should this be generic from the type or should I also include the other operations like
GetPatchFormDetailResponseDto, though I'm not sure whether this will create an issue tomorrow in any other case.

Reply:
- I was suggesting making the dto generic so that it could be used for example in a PATCH API if we want to return the form data once the form is updated.
but as you have mentioned that you're following the pattern above, the current implementation looks fine to me